### PR TITLE
add payment modal

### PR DESCRIPTION
**Pull request by victor-shagor**

### CODE REVIEW

File: src/helpers/ValidationSchemas.js [hunk 1]
```diff
@@ -120,3 +120,17 @@ export const BankingDataSchema = Yup.object().shape({
     .required('error.required')
     .matches(/^\d{10}$/, 'error.invalid.bankAcctNo'),
 })
+
+export const PaymentDataSchema = Yup.object().shape({
+  paymentPageName: Yup.string().required('error.required'),
+  amountType: Yup.string().required('error.required'),
+  amount: Yup.string().required('error.required'),
```
Comment (ikem-legend) [inline src/helpers/ValidationSchemas.js:127]: @victor-shagor You should really test your validation before pushing your code. If amount is string with no other checks, what stops me from submitting just letters as my amount? Also means you're not checking if the minimum amount is entered

File: src/helpers/ValidationSchemas.js [hunk 2]
```diff
@@ -120,3 +120,17 @@ export const BankingDataSchema = Yup.object().shape({
...
+  paymentLinkType: Yup.string().required('error.required'),
+  redirectLink: Yup.string().required('error.required'),
+  customLink: Yup.string().required('error.required'),
+  socialImage: Yup.mixed().test(
+    'fileMaxSize',
+    'error.size.fileTooLarge',
+    (file) => file && file.size <= 1000000,
```
Comment (ikem-legend) [inline src/helpers/ValidationSchemas.js:134]: @victor-shagor The size used here is the size in bytes and 1MB is 1024000 bytes not 1000000 as you wrote

File: src/components/pages/payments/NewPaymentDataForm.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            ) : null}
+          </Col>
+        </Row>
+      </FormGroup>
+      <FormGroup className="has-float-label error-l-100 mb-4">
+        <Row>
+          <Col md="6">
+            <p>Do you want to collect extra information</p>
+            <Field
+              name="extraInfo"
```
Comment (ikem-legend) [inline src/components/pages/payments/NewPaymentDataForm.js:150]: @victor-shagor extraInfo is part of your form yet it's not in your validation schema

File: src/components/pages/payments/NewPaymentDataForm.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            />
+            <label htmlFor="socialImage">
+              <p>
+                This image will show when the page is shared on social media. We
+                recommend a 1024 x 512 pixel JPG or PNG, under 1 MB in size.
+              </p>
+              <div className="text-center imgUpload">
+                <p className="mt-2">Choose image file</p>
+              </div>
+            </label>
```
Comment (ikem-legend) [inline src/components/pages/payments/NewPaymentDataForm.js:134]: @victor-shagor Fix the selected image display. There are already examples in the app in the onboarding section

File: src/components/pages/payments/NewPaymentDataForm.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+              placeholder="e.g Phone Number"
+              prepend="Name of field"
+            />
+          </Col>
+        </Row>
+      </FormGroup>
+      <Button
+        className="btn btn-primary btn-no-br float-right mt-5"
+        size="lg"
+        type="submit"
```
Comment (ikem-legend) [inline src/components/pages/payments/NewPaymentDataForm.js:161]: @victor-shagor You didn't add the disabled check for the button. I've added it in my fix anyway

File: src/components/pages/payments/NewPaymentModal.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import {Formik} from 'formik'
+import React from 'react'
+import log from 'loglevel'
+import {Modal, ModalHeader, ModalBody} from 'reactstrap'
+import {PaymentDataSchema} from '../../../helpers/ValidationSchemas'
+import PaymentDataForm from './NewPaymentDataForm'
+
+const PaymentModal = (props) => {
```
Comment (ikem-legend) [inline src/components/pages/payments/NewPaymentModal.js:8]: @victor-shagor Destructure your props directly here not below

File: src/components/pages/payments/NewPaymentDataForm.js (hunk 4 above)
Comment (victor-shagor) [reply src/components/pages/payments/NewPaymentDataForm.js]: fixed, thanks

**Pull request merged**